src/BSON/UTCDateTime.c

```diff
@@ -223,18 +217,18 @@ static PHP_METHOD(UTCDateTime, __set_state)
 static PHP_METHOD(UTCDateTime, __toString)
 {
 	php_phongo_utcdatetime_t* intern;
-	char*                     tmp;
-	int                       tmp_len;
+	char                      s_milliseconds[24];
+	int                       s_milliseconds_len;
 
 	intern = Z_UTCDATETIME_OBJ_P(getThis());
 
 	if (zend_parse_parameters_none() == FAILURE) {
 		return;
 	}
 
-	tmp_len = spprintf(&tmp, 0, "%" PRId64, intern->milliseconds);
-	PHONGO_RETVAL_STRINGL(tmp, tmp_len);
-	efree(tmp);
+	s_milliseconds_len = snprintf(s_milliseconds, sizeof(s_milliseconds), "%" PRId64, intern->milliseconds);
+
+	PHONGO_RETVAL_STRINGL(s_milliseconds, s_milliseconds_len);
```

Comment:
What is the reason for making this change? The use of ``snprintf`` is potentially unsafe due to strange ``\0`` termination rules, and ``spprintf`` is preferred.

Reply:
`jsonSerialize()`, which is effectively the same thing, was already using `snprintf()` with a static buffer (24 is sufficient to hold INT64_MIN and the null byte).

I don't follow why this would be potentially unsafe. My understanding is that `snprintf()` _always_ adds a null byte (at the expense of the string it's printing if it were to approach the buffer size). See: https://stackoverflow.com/q/7706936/162228. This makes it safer than `sprintf()` (risk of no termination) and comparable to `spprintf()`, which would also always terminate the allocated string.